### DISCUSSION TITLE
State of JS @ The Guardian

### DISCUSSION THREAD

**Discussion by SiAdcock** [guardian/csnx#426] [Team Posts]
In the CS Infra group I'm sure we're all interested in delivering _real value_ to the department and the organisation. A common question that comes up is how you measure that. 

I'd like to propose one source of metrics: a developer survey sent out to the entire department every 6 months. We assess the general level of knowledge of technologies, methodologies and resources that we care about. By analysing the results and tracking changes over time, we can ensure that our colleagues are spending their valuable time learning things that are important to Guardian engineering so that we are all pulling in the same direction. Examples include:

- How regularly do you work on client side development? (every day, a couple times per week, a couple times per month, rarely)
- How confident do you feel using React? (Very confident, not very confident)
- How confident do you feel using a screenreader for testing purposes (very confident, I can get by, I've never used a screenreader)

The [State of JS](https://2019.stateofjs.com) survey could be a useful source of starter questions.

If we track the results (anonymised) over time, we might have some indication that our group is succeeding in improving the CS experience from the perspective of our engineering colleagues.

**Comment by JamieB-gu** [comment]
I've started a [draft doc](https://docs.google.com/document/d/1cHXygYRm--_lPYmTPqyhqjnrYge2woahyl3UsoKfdHY/edit?usp=sharing) for this (in the CS Infra shared drive). Let's start filling out the "What do we want to know?" section!
